From 247294cb19dc83b384646f7a329f44a277022c91 Mon Sep 17 00:00:00 2001
From: Aikar <aikar@aikar.co>
Date: Mon, 30 Apr 2018 13:29:44 -0400
Subject: [PATCH] Enderman.teleportRandomly()

Ability to trigger the vanilla "teleport randomly" mechanic of an enderman.

diff --git a/src/main/java/net/minecraft/server/EntityEnderman.java b/src/main/java/net/minecraft/server/EntityEnderman.java
index a766b3895..0f64c8f2b 100644
--- a/src/main/java/net/minecraft/server/EntityEnderman.java
+++ b/src/main/java/net/minecraft/server/EntityEnderman.java
@@ -184,6 +184,7 @@ public class EntityEnderman extends EntityMonster {
         super.mobTick();
     }
 
+    public boolean teleportRandomly() { return dA(); } // Paper - OBFHELPER
     protected boolean dA() {
         double d0 = this.locX + (this.random.nextDouble() - 0.5D) * 64.0D;
         double d1 = this.locY + (double) (this.random.nextInt(64) - 32);
diff --git a/src/main/java/org/bukkit/craftbukkit/entity/CraftEnderman.java b/src/main/java/org/bukkit/craftbukkit/entity/CraftEnderman.java
index b81693d9f..9bc6a6c0c 100644
--- a/src/main/java/org/bukkit/craftbukkit/entity/CraftEnderman.java
+++ b/src/main/java/org/bukkit/craftbukkit/entity/CraftEnderman.java
@@ -17,6 +17,7 @@ public class CraftEnderman extends CraftMonster implements Enderman {
         super(server, entity);
     }
 
+    @Override public boolean teleportRandomly() { return getHandle().teleportRandomly(); } // Paper
     public MaterialData getCarriedMaterial() {
         IBlockData blockData = getHandle().getCarried();
         return (blockData == null) ? Material.AIR.getNewData((byte) 0) : CraftMagicNumbers.getMaterial(blockData);
-- 
2.18.0

